lib/mutgen.c

```diff
@@ -437,17 +516,19 @@ mutgen_generate(mutgen_t *self, tsk_table_collection_t *tables, int flags)
             branch_mutations = gsl_ran_poisson(self->rng, mu);
             for (l = 0; l < branch_mutations; l++) {
                 /* Rejection sample positions until we get one we haven't seen before. */
-                /* TODO add a maximum number of rejections here */
-                do {
+                for (k = 0; k < max_num_samples; k++) {
```

Comment:
What happens if we hit the maximum number of rejections?

Reply:
yes, good question - an error, i suppose

Reply:
well, we should throw an error in principle, like
```
                if ((k == max_num_rejections) && (flags & MSP_DISCRETE_SITES)) {
                    ret = MSP_ERR_INSUFFICIENT_SITES;
                    goto out;
                }
```
... but we're never going to be able to test it because to trigger it we need to actually run out of floating point numbers. Should we do this, or just throw an assert in?

Reply:
Right, there's no point in worrying about this as we'd run out of memory a thousand years before this error could happen. No point in worrying about it, so I've simplified the code.